Lib/idlelib/browser.py

```diff
@@ -161,21 +162,19 @@ def GetSubList(self):
 
     def OnDoubleClick(self):
         "Open a module in an editor window when double clicked."
-        if os.path.normcase(self.file[-3:]) != ".py":
-            return
-        if not os.path.exists(self.file):
-            return
-        file_open(self.file)
+        if self.IsExpandable() and os.path.exists(self.file):
+            file_open(self.file)
```

Comment:
For no-extension files, IsExpandable requires that the file be open.  It make no sense to test that a file exists after is has been opened.  Nor to close and reopen after the uneeded test.  Testing no-extension files should be separate.